testnet-contracts/contracts/Peggy.sol

```diff
@@ -138,55 +136,90 @@ contract Peggy is Processor {
     }
 
     /*
-     * @dev: Unlocks ethereum/erc20 tokens, called by provider.
-     *
-     *       This is a shortcut utility method for testing purposes.
-     *       In the future bidirectional system, unlocking functionality
-     *       will be guarded by validator signatures.
-     *
-     * @param _id: Unique key of the item.
-     */
-    function unlock(
-        bytes32 _id
+    * @dev: makeOracleClaimOnCosmosBridgeClaim
+    *       Processes a validator's claim on an existing CosmosBridgeClaim
+    */
+    function makeOracleClaimOnCosmosBridgeClaim(
+        uint256 _cosmosBridgeNonce,
+        bytes32 _contentHash,
+        bytes memory _signature
     )
-        external
-        onlyProvider
-        canDeliver(_id)
-        returns (bool)
+        public
+        onlyValidator()
+        isProcessing(
+            _cosmosBridgeNonce
+        )
+        returns(bool)
     {
-        require(isLocked(_id));
 
-        // Transfer item's funds and unlock it
-        (address payable sender,
-            address token,
-            uint256 amount,
-            uint256 uniqueNonce) = complete(_id);
+        // Create a new oracle claim
+        newOracleClaim(
+            _cosmosBridgeNonce,
+            msg.sender,
+            _contentHash,
+            _signature
+        );
+    }
 
-        //Emit unlock event
-        emit LogUnlock(
-            _id,
-            sender,
-            token,
-            amount,
-            uniqueNonce
+    /*
+    * @dev: processProphecyOnOracleClaims
+    *       Processes an attempted prophecy on a CosmosBridgeClaim's OracleClaims
+    */
+   function processProphecyOnOracleClaims(
+        uint256 _cosmosBridgeNonce,
+        bytes32 _hash,
+        address[] memory _signers,
```

Comment:
looks like we're expecting all signatures to be gathered offchain and posted together in a single transaction? this is fine process-wise, but not sure if fits our current relayer implementation and plans for V1, so lets discuss on call.